doc/esmvalcore/fixing_data.rst

```diff
@@ -0,0 +1,231 @@
+.. fixing_data:
+
+*************
+Dataset fixes
+*************
+
+Some (model) datasets contain (known) errors that would normally prevent them
+from being processed correctly by the ESMValTool. The errors can be in
+the metadata describing the dataset and/or in the actual data.
+Typical examples of such errors are missing or wrong attributes (e.g.
+attribute ''units'' says 1e-9 but data are actually in 1e-6), missing or
+mislabeled coordinates (e.g. ''lev'' instead of ''plev'' or missing
+coordinate bounds like ''lat_bnds'') or problems with the actual data
+(e.g. cloud liquid water only instead of sum of liquid + ice as specified by the CMIP data request).
+
+The ESMValTool can apply on the fly fixes to data sets that have
+known errors that can be fixed automatically.
+
+Fix structure
+=============
+
+Fixes are Python classes stored in ``esmvaltool/cmor/_fixes/[PROJECT]/[DATASET].py``
+that derive from :class:`esmvalcore.cmor._fixes.fix.Fix` and
+are named after the short name of the variable they fix. You can use the name
+``AllVars`` if you want the fix to be applied to the whole dataset
+
+.. warning::
+    Be careful to replace any ``-`` with ``_`` in your dataset name.
+    We need this replacement to have proper python module names.
+
+The fixes are automatically loaded and applied when the dataset is preprocessed.
+
+Fixing a dataset
+================
+
+To illustrate the process of creating a fix we are going to construct a new
+one from scratch for a fictional dataset. We need to fix a CMIP7 model
+called PERFECT-MODEL that is reporting a missing latitude coordinate for
+variable tas.
+
+Check the output
+----------------
+
+Next to the error message, you should see some info about the iris cube: size,
+coordinates. In our example it looks like this:
+
+.. code-block:: python
+
+   cube output with latitude called altitude
+
+
+So now the mistake is clear: the latitude coordinate is badly named and the
+fix should just rename it.
+
+Create the fix
+--------------
+
+We start by creating the module file. In our example the path will be
+``esmvaltool/cmor/_fixes/CMIP7/PERFECT_MODEL.py``. If it already exists
```

Comment:
Do you want to store it really in this path? If yes, I understand that you want to use CMIP7 to make sure it does not get confused with anything CMIP6 related. As soon as we get CMIP7 data, though, we would have the same problem.